From: Baoquan He <bhe@redhat.com>
Subject: mm-init-remove-the-unnecessary-special-treatment-for-memory-less-node-v2
Date: Wed, 10 Apr 2024 11:35:29 +0800

redo code comments, per Mike

As Mike suggested, the old code comments above the 'continue' statement is
still useful for easier understanding code and system behaviour.  So
rephrase and move them above line 'if (pgdat->node_present_pages)'. 
Thanks to Mike.  

Link: https://lkml.kernel.org/r/ZhYJAVQRYJSTKZng@MiWiFi-R3L-srv
Signed-off-by: Baoquan He <bhe@redhat.com>
Cc: Mel Gorman <mgorman@suse.de>
Cc: "Mike Rapoport (IBM)" <rppt@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/mm_init.c |    9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

--- a/mm/mm_init.c~mm-init-remove-the-unnecessary-special-treatment-for-memory-less-node-v2
+++ a/mm/mm_init.c
@@ -1840,7 +1840,14 @@ void __init free_area_init(unsigned long
 		pgdat = NODE_DATA(nid);
 		free_area_init_node(nid);
 
-		/* Any memory on that node */
+		/*
+		 * No sysfs hierarcy will be created via register_one_node()
+		 *for memory-less node because here it's not marked as N_MEMORY
+		 *and won't be set online later. The benefit is userspace
+		 *program won't be confused by sysfs files/directories of
+		 *memory-less node. The pgdat will get fully initialized by
+		 *hotadd_init_pgdat() when memory is hotplugged into this node.
+		 */
 		if (pgdat->node_present_pages) {
 			node_set_state(nid, N_MEMORY);
 			check_for_memory(pgdat);
_
